src/app/CommandSender.cpp

```diff
@@ -409,6 +429,27 @@ CHIP_ERROR CommandSender::ProcessInvokeResponseIB(InvokeResponseIB::Parser & aIn
         }
         ReturnErrorOnFailure(err);
 
+        if (commandRef.HasValue() && mpPendingResponseTracker != nullptr)
+        {
+            err = mpPendingResponseTracker->ResponseReceived(commandRef.Value());
+            if (err != CHIP_NO_ERROR)
+            {
+                // This can happen for two reasons:
+                // 1. The current InvokeResponse is a duplicate (based on its commandRef).
+                // 2. The current InvokeResponse is for a request we never sent (based on its commandRef).
+                // Used when logging errors related to server violating spec.
+                ScopedNodeId remoteScopedNode;
```

Comment:
```suggestion
                [[maybe_unused]] ScopedNodeId remoteScopedNode;
```
e.g. when error logging is disabled.